### ISSUE TITLE
Lack of template processor and missing methods

### ISSUE THREAD

**Issue by orkungedik** [hoeken/PsychicHttp#55]
Hi,

I am trying to convert my project from ESPAsyncWebServer to PsychicHttp for SSL support reason. By using ESPAsyncWebServer I can send response to the client by `request->send_P(200, "text/html", page_html, page_html_processor);` . So I can provide page_html_processor parameter in send_P method in order to replace variables defined in static html. But I can't find a matching method with send_P in PsychicHttp library. I found **reply** method but it hasn't a callback function parameter like AwsTemplateProcessor callback = nullptr.

Is there any way to use html template processor paramater in PsychicHttp?

Thank you,

Orkun Gedik

**Comment by orkungedik** [comment]
Additionally, implementations are missing, below;

```AsyncCallbackWebHandler& on(const char* uri, WebRequestMethodComposite method, ArRequestHandlerFunction onRequest, ArUploadHandlerFunction onUpload);
    AsyncCallbackWebHandler& on(const char* uri, WebRequestMethodComposite method, ArRequestHandlerFunction onRequest, ArUploadHandlerFunction onUpload, ArBodyHandlerFunction onBody);
```

**Comment by Chris--A** [comment]
I'm working on a solution for templating at the moment. Might have a candidate tonight.

**Comment by hoeken** [comment]
Template processor from @Chris--A is merged and live in v1.2
